### DISCUSSION TITLE
[Feature] Make 'Smart Search' smart, so it understands the intention without populating all the dedicated fields - like in Google Photos

### DISCUSSION THREAD

**Discussion by dgtal1** [immich-app/immich#8903] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I don't think Smart Search is really smart. It allows specifying a context like 'white pants', but it doesn't understand the basic parameters like location or capture date.

E.g. when I type in location, where the photos I want to see were taken - it throws me a buch of photos that are completely not related to my query.
Same as when I type in a date e.g. '2024' - I get a bunch of photos from different years and no relation to my query.
And the same when I type in a person name - I get completely unrelated results.

I know I can search by populating the fields, but sometimes it's not convenient.

I think a smart search should understand at least the basic queries related to location and date/time without populating the dedicated fields. 
Sample queries:
* 2024 - give me all photos taken in 2024
* June - give me photos taken in June any year
* June 2024 - all photos taken in June 2024
* London - give me all the photos taken in London
* Alex - I want to see all photos of Alex

Would be great if combinations were possible:
* Alex 2024
* London 2024
* London July

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
#7264

**Reply by dgtal1** [reply]
#7264 is similar, but it addresses this only partially as it requires certain tags to be used. Still not smart and you need to remember the syntax (keyword/filter names). Although would be good for the start.
